packages/vite/src/node/server/moduleGraph.ts

```diff
@@ -164,6 +183,30 @@ export class ModuleGraph {
         }
       }
     })
+    // ensure the module entry point
+    mod.importers.forEach((importerMod) => {
+      if (importerMod.entryPoint === 'self') {
```

Comment:
a question - how do circular deps affect something like this? I guess that's just a general ESM problem of which the answer is "circular deps are illegal"

Reply:
Thanks for review. I think this is no circular deps affect problem. Because mod just find the parent node (mod.importers) which is first to use the mod.